controllers/articlesController.js

```diff
@@ -311,4 +284,145 @@ export default class PostController {
+        message: "No more articles found, still need to read more?"
+      });
+    } catch (error) {
+      return res
+        .status(INTERNAL_SERVER_ERROR)
+        .json({ message: "Can't get the articles, server error" });
+    }
+  }
+
+  /**
+   * @description It helps the user to update any of his articles.
+   * @param  {object} req - The request object
+   * @param  {object} res - The response object
+   * @returns {object} It returns the request's response object
+   */
+
+  static async update(req, res) {
+    try {
+      const { slug } = req.params;
+      const { user: { id: userId } = {}, body } = req;
+      const article = await Article.findOne({ where: { slug, userId } });
+      if (article) {
+        const allowed = _.pick(body, Object.keys(articleSchema));
+        const valid = await articleValidator(
+          allowed,
+          _.pick(articleSchema, Object.keys(allowed))
+        );
+        const { tagsList = [], ...rest } = valid;
+        if (tagsList.length) {
+          tagsList.map(async t => {
+            const tag = await Tag.findOrCreate({ where: { name: t } }).spread(
+              tg => tg
+            );
+            await article.addTagsList(tag);
+          });
+        }
+        const updated = await article.update(rest, {
+          where: { id: article.id, userId },
+          returning: true,
+          limit: 1
+        });
+        const tags = await updated
+          .getTagsList({ attributes: ["name"] })
+          .map(t => t.name || null);
+        return res.status(CREATED).json({
+          article: {
+            ...updated.get(),
+            author: _.pick(await updated.getAuthor(), ["username", "bio", "image"]),
+            tagsList: tags
+          },
+          message: `Updated successfully`
+        });
+      }
+      return res.status(UNAUTHORIZED).json({
+        message: `You can only update the article you authored`
+      });
+    } catch (error) {
+      return error.details
+        ? res.status(BAD_REQUEST).json({ message: error.details[0].message })
+        : res.status(INTERNAL_SERVER_ERROR).json({
+            message: "Can't update the article, server error"
+          });
+    }
+  }
+
+  /**
+   * @description It helps the user to delete any of his articles.
+   * @param  {object} req - The request object
+   * @param  {object} res - The response object
+   * @returns {object} It returns the request's response object
+   */
+
+  static async deleteOne(req, res) {
+    try {
+      const { slug } = req.params;
+      const { user: { id: userId } = {} } = req;
+      const article = await Article.findOne({ where: { slug } });
+      if (article.userId !== userId || !article) {
+        return res.status(UNAUTHORIZED).json({
+          message: `You can only delete the article you authored`
+        });
+      }
+      await ArticleTags.destroy({ where: { articleId: article.id } });
+      await Article.destroy({ where: { id: article.id } });
```

Comment:
For auditing purpose it's not advisable to delete data in the database , 
it's advisable to add an isDeleted column to the table and make it true if you want to delete a row .
And update get all article to only retrieve article where isDeleted is false.
You can perform this in sequelize by using paranoid parameter.
You can read more about paranoid [here](http://docs.sequelizejs.com/manual/models-definition.html) in configuration section.
Thanks...

Reply:
Thanks, @espoirMur, for bringing that point in. I think that will be the best practice that will make our application stay with data for the future security concerns that may come. I'm gonna take my time to go through the resources you shared and implement it.